lib/internal/readline/interface.js

```diff
@@ -494,22 +497,29 @@ class Interface extends InterfaceConstructor {
 
     // If the trimmed line is empty then return the line
     if (StringPrototypeTrim(this.line).length === 0) return this.line;
-    const normalizedLine = this[kNormalizeHistoryLineEndings](this.line, '\n', '\r', false);
+
+    // This is necessary because each line would be saved in the history while creating
+    // A new multiline, and we don't want that.
+    if (this[kIsMultiline] && this.historyIndex === -1) {
+      ArrayPrototypeShift(this.history);
+    }
+    // If the last command errored and we are trying to edit the history to fix it
+    // Remove the broken one from the history
+    if (this[kLastCommandErrored]) {
```

Comment:
I believe this should be an else case. If the former one removes the line and this removes another one, it would actually be a bug, if I am not mistaken.

```suggestion
    else if (this[kLastCommandErrored]) {
```

Reply:
You are right, with the last change this would have been a bug!